saltproc/depcode.py

```diff
@@ -159,20 +166,346 @@ def write_mat_file(self, dep_dict, dep_end_time):
+
+    def write_depcode_input(self, reactor, dep_step, restart):
+        """ Writes prepared data into OpenMC input file(s).
+
+        Parameters
+        ----------
+        reactor : Reactor
+            Contains information about power load curve and cumulative
+            depletion time for the integration test.
+        dep_step : int
+            Current depletion time step.
+        restart : bool
+            Is the current simulation restarted?
+        """
+
+        if dep_step == 0 and not restart:
+            materials = openmc.Materials.from_xml(
+                self.template_inputfiles_path['materials'])
+            geometry = openmc.Geometry.from_xml(
+                self.template_inputfiles_path['geometry'], materials=materials)
+            settings = openmc.Settings.from_xml(
+                self.template_inputfiles_path['settings'])
+            settings.particles = self.npop
+            # settings.generations_per_batch = ??
+            settings.inactive = self.inactive_cycles
+            settings.batches = self.active_cycles + self.inactive_cycles
+        else:
+            materials = openmc.Materials.from_xml(self.iter_matfile)
+            geometry = openmc.Geometry.from_xml(
+                self.iter_inputfile['geometry'], materials=materials)
+            settings = openmc.Settings.from_xml(
+                self.iter_inputfile['settings'])
+
+        materials.export_to_xml(self.iter_matfile)
+        geometry.export_to_xml(self.iter_inputfile['geometry'])
+        settings.export_to_xml(self.iter_inputfile['settings'])
+        self.write_depletion_settings(reactor, dep_step)
+        self.write_saltproc_openmc_tallies(materials, geometry)
+        del materials, geometry, settings
+
+    def write_depletion_settings(self, reactor, current_depstep_idx):
+        """Write the depeletion settings for the ``openmc.depelete``
+        module.
+
+        Parameters
+        ----------
+        reactor : Reactor
+            Contains information about power load curve and cumulative
+            depletion time for the integration test.
+        current_depstep_idx : int
+            Current depletion step.
+
+        """
+        depletion_settings = {}
+        current_depstep_power = reactor.power_levels[current_depstep_idx]
+
+        # Get current depletion step length
+        if current_depstep_idx == 0:
+            current_depstep = reactor.dep_step_length_cumulative[0]
+        else:
+            current_depstep = \
+                reactor.dep_step_length_cumulative[current_depstep_idx] - \
+                reactor.dep_step_length_cumulative[current_depstep_idx - 1]
+
+        out_path = os.path.dirname(self.iter_inputfile['settings'])
+        depletion_settings['directory'] = out_path
+        depletion_settings['timesteps'] = [current_depstep]
+
+        operator_kwargs = {}
+
+        # Right now we'll require a path to the chain file.
+        # In the future we can make this fanciet
```

Comment:
```suggestion
        # In the future we can make this fancier
```

Reply:
Isn't this unavoidable? How would you make it fancier?

Reply:
good point.